mutiny-core/src/auth.rs

```diff
@@ -86,15 +86,15 @@ impl SigningProfile {
 }
 
 #[derive(Clone)]
-pub struct AuthManager {
+pub struct AuthManager<S: MutinyStorage> {
     profiles: Arc<RwLock<Vec<SigningProfile>>>,
     xprivkey: ExtendedPrivKey,
-    storage: MutinyStorage,
+    storage: S,
     context: Secp256k1<All>,
 }
 
-impl AuthManager {
-    pub fn new(xprivkey: ExtendedPrivKey, storage: MutinyStorage) -> Result<Self, MutinyError> {
+impl<S: MutinyStorage> AuthManager<S> {
+    pub fn new(xprivkey: ExtendedPrivKey, storage: S) -> Result<Self, MutinyError> {
         let context = Secp256k1::new();
         let mut auth_profiles = storage.get_auth_profiles()?;
```

Comment:
Since we're creating a struct here to use `dyn MutinyStorage` we'd have to do `Box<dyn MutinyStorage>` because otherwise we wouldn't know the size of `AuthManager` at compile time. Using generics is a cleaner way to do it here.